File: Arcade/TheCore/Javascript/weakNumbers.js
function divisors(n) {
    let d = 0;

    for (let i = 1; i <= n; i++)
        if (n % i == 0) d++;
    
    return d;
}

function weakness(n) {
    let lvl = 0;
    const d = divisors(n);
    
    for (let i = 1; i < n; i++)
        if (divisors(i) > d) lvl++;
        
    return lvl;
}

function weakNumbers(n) {
    const weaknessSet = []
    
    for (let i = 1; i <= n; i++) weaknessSet.push(weakness(i))
    
    const weakestNumber =
        weaknessSet.reduce((prev, acc) => prev < acc ? acc : prev);
    
    return [
        weakestNumber,
        weaknessSet.filter(i => i == weakestNumber).length
    ]
}

File: Arcade/TheCore/Java/WeakNumbers.java
import java.util.stream.IntStream;
import java.util.function.IntUnaryOperator;

final class WeakNumbers {
    
    int[] weakNumbers(int n) {
        final IntUnaryOperator divisors = x ->
            (int) IntStream.rangeClosed(1, x).filter(i -> x % i == 0).count();
        
        final int[] weaknessSet =
            IntStream.rangeClosed(1, n).map(x -> {
                final int d = divisors.applyAsInt(x);
                
                return (int) IntStream.rangeClosed(1, x - 1)
                    .filter(i -> divisors.applyAsInt(i) > d).count();
            }).toArray();
            
        final int weakestNumber = IntStream.of(weaknessSet).max().getAsInt();
        
        return new int[] {
            weakestNumber,
            (int) IntStream.of(weaknessSet)
                .filter(it -> it == weakestNumber).count()
        };
    }
}